src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -845,19 +845,29 @@ private static void DetectAndMarkAmbiguousCentralTransitiveDependencies<TItem>(T
         {
             // if a central transitive node has all parents disputed or ambiguous mark it and its children ambiguous
             int ctdCount = centralTransitiveNodes.Count;
-            for (int i = 0; i < ctdCount; i++)
+            while (true)
```

Comment:
The number of max passes here should be central transitive nodes count correct? 
Is it safer to do that instead of a while true?

Reply:
Hmm, on one hand, you are right, the maximum number of passes is equal to the count of central transitive nodes. But, on the other hand, the logic in the loop guarantees that either there is some progress being made or the loop terminates. Therefore I don't think that we need to convert the `while(true)` into a `for` loop, but if you think otherwise - I can do it.

Reply:
Sounds you're comfortable with it as is, and I'm guessing @jeffkl is as well. 

@jeffkl your call, I approved already, so it's mergable.

Reply:
I prefer the `while(true)` construct so will merge for now.  If it shows up on anyone's radar for bad perf or something we can reassess.